File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
  score = {};
  result = [];

  for (i = 0; i < name.length; i++) {
    score[name[i]] = yearning[i];
  }

  for (i = 0; i < photo.length; i++) {
    tempScore = 0;
    for (j = 0; j < photo[i].length; j++) {
      if (photo[i][j] in score) {
        tempScore = tempScore + score[photo[i][j]];
      }
    }
    result.push(tempScore);
  }

  return result;
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    score = {}
    result = []
    for i in range(0, len(name)):
        score[name[i]] = yearning[i]
    
    for i in range(0, len(photo)):
        tempScore = 0
        for j in range(0, len(photo[i])):
            if(photo[i][j] in score):
                tempScore = tempScore + score[photo[i][j]]
        result.append(tempScore)
    
    return result